src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java

```diff
@@ -19,100 +21,174 @@
+        assertEquals(2, events2.get(0).getFullDocument().getInteger("testValue"));
+    }
+
+    @Test
+    void shouldNotHandleEventsWhenListenerIsDeregisteredFromAllPartitions() throws Exception {
+        MongoCDCManager manager = new MongoCDCManager(mongoConfig);
+
+        TestChangeStreamListener listener0 = new TestChangeStreamListener();
+        manager.registerListenerToAllPartitions(listener0);
+        manager.start();
+
+        insertDocumentsToAllPartitions();
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        List<ChangeStreamDocument<Document>> events0 = listener0.getEvents();
+        assertEquals(3, events0.size());
+
+        // when
+        manager.deregisterListenerFromAllPartitions(listener0);
+
+        // then
+        var deleteResult0 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_0_ID)));
+        var deleteResult1 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_1_ID)));
+        var deleteResult2 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_2_ID)));
+        assertEquals(1, deleteResult0.getDeletedCount());
+        assertEquals(1, deleteResult1.getDeletedCount());
+        assertEquals(1, deleteResult2.getDeletedCount());
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        // only old events are available
+        List<ChangeStreamDocument<Document>> events1 = listener0.getEvents();
+        assertEquals(3, events1.size());
+    }
+
+    @Test
+    void shouldNotHandleEventsFromSelectedPartitionWhenListenerIsDeregisteredFromThisPartition() throws Exception {
+        MongoCDCManager manager = new MongoCDCManager(mongoConfig);
+
+        TestChangeStreamListener listener0 = new TestChangeStreamListener();
+        manager.registerListenerToAllPartitions(listener0);
+        manager.start();
+
+        insertDocumentsToAllPartitions();
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        List<ChangeStreamDocument<Document>> events0 = listener0.getEvents();
+        assertEquals(3, events0.size());
+
+        // when
+        manager.deregisterListener(listener0, List.of(0));
+
+        // then
+        var deleteResult0 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_0_ID)));
+        var deleteResult1 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_1_ID)));
+        var deleteResult2 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_2_ID)));
+        assertEquals(1, deleteResult0.getDeletedCount());
+        assertEquals(1, deleteResult1.getDeletedCount());
+        assertEquals(1, deleteResult2.getDeletedCount());
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        // only old events and deletes from not deregistered partitions are available
+        List<ChangeStreamDocument<Document>> events1 = listener0.getEvents();
+        assertEquals(5, events1.size());
```

Comment:
5 is a magic number here which depends on:
* events0.size
* number of deleteOne calls
* size of list used for deregisterListener call (more or less)

Reply:
that's why there is a comment
change a little bit. Is it more clear now?

Reply:
LGTM